include/pq_flash_index.h

```diff
@@ -174,6 +178,12 @@ namespace diskann {
     // closest centroid as the starting point of search
     float *centroid_data = nullptr;
 
+    unsigned    num_starting_points;
+    std::string selection_strategy_of_starting_points;
+    std::unique_ptr<T[]> starting_points_data;
+    size_t                starting_points_data_dim;
```

Comment:
Why is this dimension variable necessary. Would it be the same as other vectors?

Reply:
because there are several "dim" variables.  I will change to use "data_dim"